policy/compiler.go

```diff
@@ -248,22 +278,28 @@ func CompileRule(env *cel.Env, p *Policy, opts ...CompilerOption) (*CompiledRule
 			c.env = env
 		}
 	}
-	return c.compileRule(p.Rule(), c.env, iss)
+	return c.CompileRuleImpl(p.Rule(), p, c.env, iss)
 }
 
-type compiler struct {
+// Compiler holds the state of the policy compiler.
+type Compiler struct {
```

Comment:
Prefer using functional options to configure the fields you're looking for rather than exporting the struct.